python/pyarrow/fs.py

```diff
@@ -263,7 +263,10 @@ def get_file_info_selector(self, selector):
 
     def create_dir(self, path, recursive):
         # mkdir also raises FileNotFoundError when base directory is not found
-        self.fs.mkdir(path, create_parents=recursive)
+        try:
+            self.fs.mkdir(path, create_parents=recursive)
+        except FileExistsError:
```

Comment:
I found the "FileExistsError" a bit strange, because I first thought this was about a *directory* that already existed (but apparently, fsspec's `mkdir` is fine with that as well and will pass silently if the target directory already exists, at least I checked this for the local filesystem). But so it is actually about a *file* already existing with that name. 
And the current behaviour of our filesystem is then to silently not create a directory. But is that our desired behaviour? I would say that `create_dir` should guarantee that the target directory exists after calling that method (or otherwise error)

Reply:
> But so it is actually about a file already existing with that name.

Are you sure that's the case?

Reply:
```
from fsspec.implementations.local import LocalFileSystem as FSSpecLocalFileSystem
fs = FSSpecLocalFileSystem()
fs.mkdir("test_dir")
fs.touch("test_file")

# creating an existing dir again is fine
In [9]: fs.mkdir("test_dir")

# creating a dir that already exists as a file errors
In [10]: fs.mkdir("test_file")
...
FileExistsError: [Errno 17] File exists: '/home/joris/scipy/test_file'
```

(now, this was a manual test of the expected behaviour, will check what actually happens in the failing test)

Reply:
Ah, but it's the MemoryFileSystem that fails with the exact same example from above .. So they are again not very consistent, that seems a bug in the MemoryFileSystem. Will open an issue about it.